handlers/project.go

```diff
@@ -17,24 +18,53 @@ func setupLog() {
 // GetProject handles GET requests to retrieve all projects
 func GetProject(w http.ResponseWriter, r *http.Request) {
 	setupLog()
-	queryParams := r.URL.Query()
+	vars := mux.Vars(r)
 
 	// Read a specific parameter
-	projectID := queryParams.Get(constants.ID)
+	projectID := vars["Id"]
 
 	// Use the parameters as needed
 	//TODO: return the http status code from resource
-	response, err := project.Read(r.Context(), &project.Model{Id: &projectID})
-	if err != nil {
-		_, _ = logger.Debugf("CreateProjectHandler error:%s", err.Error())
+	response := project.Read(r.Context(), &project.Model{Id: &projectID})
+	w.Header().Set("Content-Type", "application/json")
+	res, _ := json.Marshal(response)
+	if response.HttpError != "" {
+		_, _ = logger.Debugf("CreateProjectHandler error:%s", response.HttpError)
+		//http.Error(w, response.HttpError, http.StatusNotFound)
+		_, _ = w.Write(res)
 		return
 	}
-	res, err := json.Marshal(response)
+
```

Comment:
I'd suggest return response as is.. in case of statusnotfound that could be handled in place of error check in line no. 31.
Keep error handling as generic method since it will be the same bit of code in every call.